env_spec.py

```diff
@@ -1,53 +1,82 @@
 import re
 
-
-def check_input(field_array):
+valid_types_array = [
+    "color",
+    "date",
+    "datetime-local",
+    "email",
+    "month",
+    "number",
+    "password",
+    "tel",
+    "text",
+    "time",
+    "url",
+    "week",
+]
+
+
+def check_input(field_type_array):
     """
     Checks syntax of input(uppercase latin chars,numeric digits, underscores).
     Returns False for error or True for correct syntax.
     """
     alphanumeric_that_does_not_start_with_digit = r"^[A-Z_][0-9A-Z_]*$"
 
-    for line in field_array:
-        ret = re.match(alphanumeric_that_does_not_start_with_digit, line)
+    for line in field_type_array:
+        ret = re.match(alphanumeric_that_does_not_start_with_digit, line[0])
 
         if ret is None:
-            # print("SYNTAX ERROR:WRONG SYMBOLS!")
             return False
 
     return True
 
 
-def render_env_var_spec(env_var_name):
+def check_type(input_array):
+    """
+    checks if given type is valid and returns True, else returs False
+    """
+    for type in input_array:
+        if type[1] not in valid_types_array:
+            return False
+    return True
+
+
+def render_env_var_spec(env_var):
+    env_var_name = env_var[0]
+    env_var_type = env_var[1]
     env_var_name_lower = env_var_name.lower()
 
     ret_str = f'<label for="env_spec_{env_var_name_lower}"> {env_var_name}</label>'
     ret_str += "\n"
-    ret_str += (
-        f'<input id="env_spec_{env_var_name_lower}" name="{env_var_name_lower}" />'
-    )
+    ret_str += f'<input id="env_spec_{env_var_name_lower}" name="{env_var_name_lower}" type="{env_var_type}" />'
 
     return ret_str
 
 
-def render_env_spec_to_html(field_array):
-    if check_input(field_array):
+def render_env_spec_to_html(field_type_array):
```

Comment:
Can you change `render_env_spec_to_html` to accept string input, instead of a list?

If you would like to keep the "list" input implementation, you can rename this to something like `render_env_spec_lines_to_html`, implement `render_env_spec_to_html` from scratch and use `render_env_spec_lines_to_html` there.

Reply:
Also, can you please rename `array` references to `list`? There are no "arrays" in Python 😁.